wandb/sdk/launch/create_job.py

```diff
@@ -0,0 +1,430 @@
+        return None
+
+    ref = GitHubReference.parse(path)
+    if not ref:
+        wandb.termerror("Could not parse git URI")
+        return None
+
+    ref.fetch(tempdir)
+
+    commit = git_hash
+    if not commit:
+        if not ref.commit_hash:
+            wandb.termerror("Could not find git commit hash")
+            return None
+        commit = ref.commit_hash
+
+    local_dir = os.path.join(tempdir, ref.directory or "")
+    src_dir = ref.directory or ""
+    filename = ref.file
+    if not filename and path.endswith(".py"):
+        filename = os.path.basename(path)
+
+    python_version = runtime
+    if not python_version:
+        if os.path.exists(os.path.join(local_dir, "runtime.txt")):
+            with open(os.path.join(local_dir, "runtime.txt")) as f:
+                python_version = f.read().strip()
+        elif os.path.exists(os.path.join(local_dir, ".python-version")):
+            with open(os.path.join(local_dir, ".python-version")) as f:
+                python_version = f.read().strip().splitlines()[0]
+        else:
+            major, minor = get_current_python_version()
+            python_version = f"{major}.{minor}"
+
+    python_version = _clean_python_version(python_version)
+
+    # check if entrypoint is valid
+    if entrypoint:
+        if not os.path.exists(os.path.join(local_dir, entrypoint)):
+            wandb.termerror(f"Invalid entrypoint: {entrypoint}")
+            return None
+        # prepend working dir (specified in path)
+        entrypoint = os.path.join(src_dir, entrypoint)
+    elif not filename:
+        if path.endswith(".py"):
+            wandb.termerror("Invalid entrypoint provided in path")
+        else:
+            wandb.termerror(
+                "Invalid entrypoint. Provide an entrypoint in the path or use -E"
+            )
+        return None
+    else:  # try to make one from the path, assumes working dir is parent of file
+        rel_entrypoint = os.path.join(src_dir, filename)
+        if rel_entrypoint.startswith("/"):
+            rel_entrypoint = rel_entrypoint[1:]
+
+        if not os.path.exists(os.path.join(tempdir, rel_entrypoint)):
+            wandb.termerror(
+                f"Entrypoint {rel_entrypoint} not valid, specify one in the path or use -E"
+            )
+            return None
+        entrypoint = rel_entrypoint
+
+    # check if requirements.txt exists
+    if not os.path.exists(os.path.join(local_dir, "requirements.txt")):
+        repo_formd = path.replace(entrypoint, "")
+        wandb.termerror(
+            f"Could not find requirements.txt file in git repo at: {repo_formd}/requirements.txt"
+        )
+        return None
+
+    metadata = {
+        "git": {
+            "commit": commit,
+            "remote": ref.url,
+        },
+        "root": ref.repo,
+        "codePath": entrypoint,
+        "entrypoint": [f"python{python_version}", entrypoint],
+        "python": python_version,  # used to build container
+    }
```

Comment:
@KyleGoyette  Was this what you were thinking? I believe it is now working, but clearly still not optimal.